src/types/resources.d.ts

```diff
@@ -67,6 +67,22 @@ export interface GpuCard {
   vendor?: string;
   vendor_id?: string;
 }
+export interface ProxyCard {
+  nat?: boolean;
+  bind?: string;
+  proxy_protocol?: boolean;
+  listen?: {
+    type?: string;
+    mode?: number;
+    address?: string;
+    port?: string;
+  };
+  connect?: {
+    type?: string;
+    address?: string;
+    port?: string;
+  };
+}
```

Comment:
I think we don't need this type. To translate the listen/connect strings into the nested object, there is a better strategy. As discussed, similar to the memory selection, we can have a component that receives the composed listen/connect string and breaks it down on render. On change, that component can rebuild the string and only communicate the recomposed string back up to the form state.

Reply:
If we get rid of this type, this breaks things in NewProxyBtn (Previously SelectProxyBtn) and ProxyDeviceForm...